high_card.rb

```diff
@@ -1,5 +1,46 @@
-# Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+# Use these two arrays to generate a deck of cards
+  ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]
+  suits = ["hearts", "spades", "diamonds", "clubs"]
+
+  deck = ranks.product(suits).shuffle
+
+#   deck = []
+#
+#   ranks.each do |rank|
+#     suits.each do |suit|
+#     deck << [rank, suit]
+#   end
+# end
+
 
 players = []
+puts players.length.to_s + "Player so far"
+puts "Enter a player name or type play: "
+player = gets.chomp
+
+while player != "play" do
+  players << player
+  puts "Enter a player name or type play: "
+  player = gets.chomp
+end
```

Comment:
What you have here is great and works fine. A slightly cleaner version might take advantage of the `break` command which allows us to DRY up the code a little.
```rb
while true
  puts "#{players.length} so far. Enter a player name, or type 'play':"
  name = gets.chomp

  break if name == "play"
  players.push(name)
end
```